packages/principles-core/src/runtime-v2/store/sqlite-connection.ts

```
@@ -56,16 +56,28 @@ export class SqliteConnection {
+        // Verify they were applied correctly
+        const journalMode = this.db.pragma('journal_mode', { simple: true });
+        if (String(journalMode).toLowerCase() !== 'wal') {
           throw new PDRuntimeError(
             'storage_unavailable',
-            `Failed to set WAL journal mode (got: ${walResult})`,
+            `Failed to set WAL journal mode (got: ${journalMode})`,
+          );
+        }
+
+        const foreignKeys = this.db.pragma('foreign_keys', { simple: true });
+        if (!Boolean(foreignKeys)) {
```

ESLint

[error] Redundant Boolean call.

(no-extra-boolean-cast)